tests/unit/test_scheduler.py

```diff
@@ -152,3 +152,32 @@ def test_check_startup_opts(
     with pytest.raises(SchedulerError) as excinfo:
         Scheduler._check_startup_opts(mocked_scheduler)
     assert(err in str(excinfo))
+
+from types import SimpleNamespace
+
+
+@pytest.mark.parametrize(
+    'scp, fcp, err',
+    [
+        (2, 1, True),
+        (1, 1, True),
+        (1, 2, False),
+        ('19840222T0400Z', '19831213T0530Z', True),
+        ('19831213T0530Z',  '19831213T0530Z', True),
+        ('19831213T0530Z','19840222T0400Z', False)
+    ]
+)
+def test_warnif_stopcp_gte_finalcp(caplog, scp, fcp, err):
+    """Scheduler warns is stopcp ≥ finalcp"""
+    mocked_scheduler = Mock()
+    mocked_scheduler.options = SimpleNamespace(stopcp=scp, fcp=fcp)
+    mocked_scheduler.config = SimpleNamespace(cycling_type='integer')
+    
+    Scheduler.warnif_stopcp_gte_finalcp(mocked_scheduler)
+    if err == True:
+        assert caplog.records[0].message == (
+            f'Stop Cycle point \'{scp}\' will not do anything: It is the same '
+            f'or later than the final Cycle point \'{fcp}\'.'
```

Comment:
Maybe here we don't need to check the exact string is logged? Either just assert the log function has been called and the message is not empty, or that it contains the stop cycle point and the final cycle point?

Reply:
Agree. Lazy writing on my part ⇒ fragile test.